src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1114,6 +1114,11 @@ public sealed class ValidatePatternAttribute : ValidateEnumeratedArgumentsAttrib
         public RegexOptions Options { set; get; } = RegexOptions.IgnoreCase;
 
         /// <summary>
+        /// Gets or sets the custom error message that is displayed to the user
+        /// </summary>
+        public string ErrorMessage { get; set; }
```

Comment:
If you mean all **public** validation attributes, then maybe, but I'm not fully convinced.

With a custom error message, it's possible for a cmdlet author to provide a worse error message than the default, especially without localized messages.

For a few of the attributes, I can't convince myself it's worth it. And there are minor costs - an extra pointer that is never used isn't free - it requires more memory and extra time in garbage collection checking if the (usually null) value needs collecting.

`ValidateNotNullOrEmpty` stands out as one example where I wouldn't bother.  `ValidateDrive` is also probably not worth it.

And last - I can see a custom error message being useful on  `ValidateRange`, e.g. to say the value should be a positive value or whatever, but if we go that route, we need localization support similar to `HelpMessageBaseName` and `HelpMessageResourceId` in `ParameterAttribute`.